### PR TITLE
WIP: Fix #1035

### LINKED ISSUE #1035: Name resolution can fail in the simulator depending on operator order
I found a strange edge case in which name resolution of an imported operator incorrectly fails, but only if the operators are imported in a specific order and the main module is loaded in a specific way :woman_shrugging: 

This issue is light on details because I've opened a draft PR with a failing integration test in #1036.

### PR DESCRIPTION
**Pull request by shonfeder**
At the moment this is just providing a starting place for whomever works this bug.

-------

- [ ] Tests added for any new code
- [ ] Documentation added for any new functionality
- [ ] Entries added to the respective `CHANGELOG.md` for any new functionality
- [ ] Feature table on [`README.md`](../README.md#roadmap) updated for any listed functionality

### CODE REVIEW

File: quint/cli-tests.md [hunk 1]
```diff
@@ -394,3 +394,13 @@ fi
...
+
+### Regression test for #1035
+
+Reported in https://github.com/informalsystems/quint/issues/1035, the spec
+tested here was producing flaky, incorrect name resolution errors, that were
+sensitive to the order of operator and module definitions.
+
+<!-- !test exit 0 -->
+<!-- !test check regression test for 1035 -->
+    quint -r ./testFixture/simulator/importedOpOrderRegression.qnt::importedOpOrderRegression
```
Comment (shonfeder) [inline quint/cli-tests.md:406]: Name resolution is not failing on `parse` or `typecheck`, only with `-r` or `quint test testFixture/simulator/importedOpOrderRegression.qnt` (and I assume also with `run`).

**Comment by konnov** [comment]
Is this PR ready for review? It still has "WIP" in the title

**Comment by shonfeder** [comment]
This was meaning to provide, initially, just a regression test for the bug reported on #1035. I'm surprised that the integration tests are succeeding.

**Comment by bugarela** [comment]
I was not able to reproduce this locally either. Should we close this PR?

**Pull request closed without merging**